py/nlrthumb.c

```diff
@@ -71,7 +79,7 @@ __attribute__((naked)) unsigned int nlr_push(nlr_buf_t *nlr) {
         "str    lr, [r0, #8]        \n" // store lr into nlr_buf
         #endif
 
-        #if !defined(__thumb2__)
+        #if !defined(__thumb2__) && MICROPY_NLR_THUMB_FORCE_LONG_JUMP
```

Comment:
I suggest calling the option `MICROPY_NLR_THUMB_USE_LONG_JUMP`, and removing the `defined(__thumb2__)` bit from the logic.  Ie it would simply be:
```c
#if MICROPY_NLR_THUMB_USE_LONG_JUMP
...
```
Then the default is to just do a standard b-instruction for the branch, and if anyone runs into linker issues they can just enable the option.

Reply:
Hello @dpgeorge, thank you for your reply! I hope I'll get around to it soon.

Reply:
@dpgeorge done, and tested with musl-libc and glibc

Reply:
Thanks for updating, and for testing.